### ISSUE TITLE
Problem with git lfs push when manually forking a repository

### ISSUE THREAD

**Issue by clembou** [git-lfs/git-lfs#1765]
Hi,

I am attempting to fork one of my organisation's repository hosted on Github. Since the fork would be hosted in the same organisation I cannot use the built in "fork" button and have to to it manually by creating new repo and pushing to it: `git remote add newrepo https://github.com/myorg/newrepo`

When I try to push to the new repo, the push fails with the following message:

```
git lfs push smart master
Trying to push "ApplicationA.exe" with OID 04f25016390e616f65a193b9ff5d54498843dbe6856b198a2d28c512d3d1e1b3.
Not found in E:\repos\myrepo\.git\lfs\objects\04\f2.
```

git lfs ls-files says:
```
7a15ce9c98 * ApplicationA.exe
32922f60d2 * ApplicationB.exe
e53d559fd2 * Antlr/Antlr3.Runtime.Debug.dll
ab2a9e759c * Antlr/Antlr3.Runtime.JavaExtensions.dll
8c073bb62e * Antlr/Antlr3.Runtime.dll
c2a62167a5 * Antlr/Antlr3.Runtime.pdb
0acc794713 * Antlr/Antlr3.exe
f146e8a15b * Antlr/Antlr4.StringTemplate.Visualizer.dll
ab0b54b953 * Antlr/Antlr4.StringTemplate.dll
730dfd1d3a * Antlr/AntlrBuildTask.dll
10f0d0c671 * Antlr/Targets/Antlr3.Targets.CSharp2.dll
b110213cf6 * Antlr/Targets/Antlr3.Targets.CSharp3.dll
```

Basically, the current latest versions of all tracked files are correctly checked out.

```
git lfs fsck
Object ApplicationA.exe (f15d4e351e5bc584530ba657d31d7cb952311e97be2ff73e2ac07f142c849d74) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (7e529e5af2afb50a96c9181641cba727eb702b039c9c25e56a80830e2cd758ab) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (f0b3e07b7cd824978d625bc2a68097ad9397d4ca7074d664672bfc9e6cafd2ed) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (2b2dad0d7b05dcc5f2b824cabfbb66b53071c074a1f449e0c3055135ddf1e16f) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (586775977e25878d22bcc11d23d00d6c659619a81a73d21c61e20301a03696d2) could not be checked: The system cannot find the path specified.
Object ApplicationA.exe (a3a6ef33c53e92024fbee0f44c651b3ea461273f9b00c3ae561485bc338a902e) could not be checked: The system cannot find the path specified.
Object ApplicationA.exe (1ab172540bcf9f9dfb88621ebdc584578d95bf74d890403e745d994f29b2d41f) could not be checked: The system cannot find the path specified.
Object ApplicationA.exe (c042ab6774f367571e671b6b299c57a66a3ffc1bce8164d675a4b94446479cb9) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (03d59814cf1b078ec1731eba6c3156f83784eebf7476eaeadcdaacbf8ae42424) could not be checked: The system cannot find the path specified.
Object ApplicationA.exe (04f25016390e616f65a193b9ff5d54498843dbe6856b198a2d28c512d3d1e1b3) could not be checked: The system cannot find the file specified.
Object ApplicationA.exe (f509c3ed15a9e5ea748ddaf3e9c0d79b32d22e6c01d4bdc1342104fd158d0b60) could not be checked: The system cannot find the path specified.
Object ApplicationB.exe (a53b2b6ef844bb9c62e210bbbd38e255b2161f2fbd42cddce398d8759fa4dc95) could not be checked: The system cannot find the path specified.
```

But I do not have all the previous versions locally.

I would have thought that git lfs should be able to deal with this problem? e.g. fetch missing files if needed?

Alternatively, is there an easy way from me to manually download the previous revisions so that the push succeeds?

Forking the repo under my username works perfecly but since this is a work repository I need to keep the fork in the same organisation so this is not an option.

Environment details below:
```
$ git lfs env
git-lfs/1.5.3 (GitHub; windows amd64; go 1.7.4; git 23455e48)
git version 2.11.0.windows.1

LocalWorkingDir=
LocalGitDir=
LocalGitStorageDir=
LocalMediaDir=
LocalReferenceDir=
TempDir=C:\Users\CBOUSC~1\AppData\Local\Temp\git-lfs
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
BatchTransfer=true
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_LFS_PATH=C:\Program Files\Git LFS
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Comment by clembou** [comment]
I should add that I don't particularly care about the old version of these files, so it would ok for me to push the history without the lfs objects if that is an option. I just need the latest version in this case.

**Comment by ttaylorr** [comment]
Hi @clembou, thanks for opening this. From the description, it sounds like you don't have all of the LFS objects in your local copy of the repository. You can use the LFS command `fetch` to obtain a copy of all objects in your repository.

If you care about history:

```sh
$ git lfs fetch --all <remote>
```

or if you don't:

```sh
$ git lfs fetch <remote>
```

After that, your `.git/lfs/objects` directory will contain the missing objects. You should then be able to add another remote and push at will 👍.

There is some confusion going on here in that `git lfs ls-files` prints LFS _pointers_ in your working tree, which may or may not be _objects_. For instance, if I do a LFS clone without downloading any of the objects, I'll still get a long list of pointers in `ls-files`, even though I don't have the object data.

**Comment by clembou** [comment]
Thanks @ttaylorr , `--all` is what I was missing! I did run `git lfs fetch` before and definitely had the LFS objects referenced in the current `HEAD` but that is not enough to be able to push the existing history to a *new* repo.

As a quick note, I ran into this issue because I though I could just follow the normal git workflow to fork my repo e.g. 
```
git (lfs) clone myrepo 
cd myrepo
git remote add newremote
git (lfs) push --all newremote
```

It makes sense that old lfs objects are not downloaded by default but perhaps `git lfs push` could warn about this when objects are missing?

anyway, thanks for your help.
